dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-email/src/main/java/org/apache/dolphinscheduler/plugin/alert/email/MailSender.java

```diff
@@ -105,20 +105,29 @@ public MailSender(Map<String, String> config) {
 
         enableSmtpAuth = config.get(MailParamsConstants.NAME_MAIL_SMTP_AUTH);
 
-        mailUser = config.get(MailParamsConstants.NAME_MAIL_USER);
-        requireNonNull(mailUser, MailParamsConstants.NAME_MAIL_USER + mustNotNull);
+        if (Boolean.TRUE.toString().equalsIgnoreCase(enableSmtpAuth)) {
```

Comment:
I'm not quite sure about this feature since not using a password is a very unsafe operation. I think we should not provide this way to users. WDYT? @ruanwenjun @caishunfeng

Reply:
Yes, this is unsafe, but I think we can support this. In my knowledge, DS should directly use the email protocol to send mail, if the mail server/protocol support send/receive mail without password, then we also need to support this.

Reply:
> Yes, this is unsafe, but I think we can support this. In my knowledge, DS should directly use the email protocol to send mail, if the mail server/protocol support send/receive mail without password, then we also need to support this.

agree with @ruanwenjun :+1: